typescript/cli/src/read/warp.ts

```diff
@@ -22,82 +24,83 @@ export async function runWarpRouteRead({
+    logRed(
+      'Invalid input parameters. Please provide either a token symbol or both chain name and token address',
+    );
+    process.exit(1);
+  }
 
-  let addresses: ChainMap<string>;
-  if (symbol || warp) {
-    const warpCoreConfig =
-      context.warpCoreConfig ?? // this case is be handled by MultiChainHandler.forWarpCoreConfig() interceptor
-      (await getWarpCoreConfigOrExit({
+  const warpCoreConfig = hasWarpConfig
+    ? await getWarpCoreConfigOrExit({
         context,
-        warp,
         symbol,
-      }));
-
-    // TODO: merge with XERC20TokenAdapter and WarpRouteReader
-    const xerc20Limits = await Promise.all(
-      warpCoreConfig.tokens
-        .filter(
-          (t) =>
-            t.standard === TokenStandard.EvmHypXERC20 ||
-            t.standard === TokenStandard.EvmHypXERC20Lockbox,
-        )
-        .map(async (t) => {
-          const provider = multiProvider.getProvider(t.chainName);
-          const router = t.addressOrDenom!;
-          const xerc20Address =
-            t.standard === TokenStandard.EvmHypXERC20Lockbox
-              ? await HypXERC20Lockbox__factory.connect(
-                  router,
-                  provider,
-                ).xERC20()
-              : await HypXERC20__factory.connect(
-                  router,
-                  provider,
-                ).wrappedToken();
-
-          const xerc20 = IXERC20__factory.connect(xerc20Address, provider);
-          const mint = await xerc20.mintingCurrentLimitOf(router);
-          const burn = await xerc20.burningCurrentLimitOf(router);
-
-          const formattedLimits = objMap({ mint, burn }, (_, v) =>
-            ethers.utils.formatUnits(v, t.decimals),
-          );
-
-          return [t.chainName, formattedLimits];
-        }),
-    );
+      })
+    : undefined;
 
-    if (xerc20Limits.length > 0) {
-      logGray('xERC20 Limits:');
-      logTable(Object.fromEntries(xerc20Limits));
-    }
+  const addresses = warpCoreConfig
+    ? Object.fromEntries(
+        warpCoreConfig.tokens.map((t) => [t.chainName, t.addressOrDenom!]),
+      )
+    : { [chain!]: address! };
 
-    addresses = Object.fromEntries(
-      warpCoreConfig.tokens.map((t) => [t.chainName, t.addressOrDenom!]),
-    );
-  } else if (chain && address) {
-    addresses = {
-      [chain]: address,
-    };
-  } else {
-    logRed(`Please specify either a symbol, chain and address or warp file`);
-    process.exit(1);
+  return deriveWarpRouteConfigs(context, addresses, warpCoreConfig);
+}
+
+export async function getWarpRouteConfigsByCore({
+  context,
+  warpCoreConfig,
+}: {
+  context: CommandContext;
+  warpCoreConfig: WarpCoreConfig;
+}): Promise<ChainMap<HypTokenRouterConfig>> {
+  const addresses = Object.fromEntries(
+    warpCoreConfig.tokens.map((t) => [t.chainName, t.addressOrDenom!]),
+  );
+
+  return deriveWarpRouteConfigs(context, addresses, warpCoreConfig);
+}
+
+async function deriveWarpRouteConfigs(
```

Comment:
A better name would be `deriveWarpDeployConfigs` because what we read from the `EvmERC20WarpRouteReader` is a deployment config.

In general, we should steer our naming towards a smaller set of terms, avoiding synonyms. My understanding is that we want to call the two types of warp route configs as "core" and "deploy" -- please correct me if I'm wrong here.